opengcs/go.mod

```diff
@@ -0,0 +1,24 @@
+module github.com/Microsoft/opengcs
```

Comment:
Yeah, do we want to keep this or just move everything to the main go.mod file in hcsshim?

Reply:
This PR was mostly just getting the code and history in here and we could actually place things where they belong/delete things shortly after. If we'd rather do everything all in here that's fine with me too, it'll just be harder to review